nomad/job_endpoint_test.go

```diff
@@ -6858,87 +6873,103 @@ func TestJobEndpoint_Dispatch(t *testing.T) {
-				case "":
-					if !tc.noEval {
-						t.Fatalf("Bad response")
-					}
-				default:
-					if tc.noEval {
-						t.Fatalf("Got eval %q", dispatchResp.EvalID)
-					}
-				}
+			if dispatchResp.DispatchedJobID == "" {
+				t.Fatalf("Bad response")
+			}
 
-				if dispatchResp.DispatchedJobID == "" {
-					t.Fatalf("Bad response")
-				}
+			state := s1.fsm.State()
+			ws := memdb.NewWatchSet()
+			out, err := state.JobByID(ws, tc.parameterizedJob.Namespace, dispatchResp.DispatchedJobID)
+			must.NoError(t, err)
+			if err != nil {
+				t.Logf("err: %v", err)
+			}
 
-				state := s1.fsm.State()
-				ws := memdb.NewWatchSet()
-				out, err := state.JobByID(ws, tc.parameterizedJob.Namespace, dispatchResp.DispatchedJobID)
-				if err != nil {
-					t.Fatalf("err: %v", err)
-				}
-				if out == nil {
-					t.Fatalf("expected job")
-				}
-				if out.CreateIndex != dispatchResp.JobCreateIndex {
-					t.Fatalf("index mis-match")
-				}
-				if out.ParentID != tc.parameterizedJob.ID {
-					t.Fatalf("bad parent ID")
-				}
-				if !out.Dispatched {
-					t.Fatal("expected dispatched job")
-				}
-				if out.IsParameterized() {
-					t.Fatal("dispatched job should not be parameterized")
-				}
-				if out.ParameterizedJob == nil {
-					t.Fatal("parameter job config should exist")
-				}
+			must.NotNil(t, out)
+			if out == nil {
+				t.Logf("expected job, got nil")
+			}
 
-				// Check that the existing job is returned in the case of a supplied idempotency token
-				if tc.idempotencyToken != "" && tc.existingIdempotentJob != nil {
-					if dispatchResp.DispatchedJobID != initialIdempotentDispatchResp.DispatchedJobID {
-						t.Fatal("dispatched job id should match initial dispatch")
-					}
+			must.Eq(t, out.CreateIndex, dispatchResp.JobCreateIndex)
+			if out.CreateIndex != dispatchResp.JobCreateIndex {
+				t.Logf("index mis-match")
+			}
 
-					if dispatchResp.JobCreateIndex != initialIdempotentDispatchResp.JobCreateIndex {
-						t.Fatal("dispatched job create index should match initial dispatch")
-					}
-				}
+			must.Eq(t, out.ParentID, tc.parameterizedJob.ID)
+			if out.ParentID != tc.parameterizedJob.ID {
+				t.Logf("bad parent ID")
+			}
 
-				if tc.noEval {
-					return
-				}
+			must.Eq(t, out.Dispatched, true)
+			if !out.Dispatched {
+				t.Logf("expected dispatched job")
+			}
 
-				// Lookup the evaluation
-				eval, err := state.EvalByID(ws, dispatchResp.EvalID)
-				if err != nil {
-					t.Fatalf("err: %v", err)
-				}
+			must.Eq(t, out.IsParameterized(), false)
+			if out.IsParameterized() {
+				t.Logf("dispatched job should not be parameterized")
+			}
+			must.NotNil(t, out.ParameterizedJob)
+			if out.ParameterizedJob == nil {
+				t.Logf("parameter job config should exist")
+			}
 
-				if eval == nil {
-					t.Fatalf("expected eval")
-				}
-				if eval.CreateIndex != dispatchResp.EvalCreateIndex {
-					t.Fatalf("index mis-match")
+			if tc.priority != 0 {
+				must.Eq(t, tc.priority, out.Priority)
+			}
```

Comment:
I'd make sure that `tc.priority` is always set (ex. it could be `tc.expectPriority`), so that we can see it at a glance in the test case list and then there's no conditional here -- we just always call `must.Eq`.